src/farbtastic.js

```diff
@@ -490,7 +490,7 @@ $._farbtastic = function (container, options) {
   };
 
   // Parse options.
-  if (!options.callback) {
+  if (!options.callback && !options.width) {
     options = { callback: options };
```

Comment:
This should be handled differently. If there is no callback property than one should be defined rather than redefining options.